### PR TITLE
Update handling of state for QuestionCreate and QuestionUpdate

### PR DESCRIPTION
**Pull request by cliffpyles**
Update handling of state for QuestionCreate and QuestionUpdate containers. Changed the state to use a reducer and handle all state in the container. The data now flows down through the props and execute handlers that are passed through the props.

The commit also introduces the QuestionForm component, which was formally specific to the Questions containers.

### CODE REVIEW

**Comment by megsachdev** [comment]
@cliffpyles Is there a reason why this was suggested in the last PR and done as a part of new PR?

**Comment by cliffpyles** [comment]
@megha I'm not sure I understand the question. Do you mean why was it not done in the last PR?

**Comment by megsachdev** [comment]
Looks great.

File: src/components/QuestionForm/QuestionFormChoice.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import TextField from 'components/TextField'
+import Button from 'components/Button'
+import { TrashIcon } from 'components/Icon'
+
+export default ({ _key, value, onRemoveChoice, onUpdateChoice }) => {
+  return (
+    <TextField
```
Comment (megsachdev) [inline src/components/QuestionForm/QuestionFormChoice.js]: @cliffpyles just wanted to know how will the user select the correct answer?

**Pull request merged**